modules/UI/shared_video/SharedVideo.js

```diff
@@ -421,20 +438,63 @@ export default class SharedVideoManager {
      * @param {boolean} indicates if this mute was a result of user interaction,
      * i.e. pressing the mute button or it was programatically triggerred
      */
-    localAudioMuted (muted, userInteraction) {
+    onLocalAudioMuted (muted, userInteraction) {
         if(!this.player)
             return;
 
         if (muted) {
             this.mutedWithUserInteraction = userInteraction;
-            return;
         }
+        else if (this.player.getPlayerState() !== YT.PlayerState.PAUSED) {
+            this.mutePlayer(true);
+        }
+    }
+
+    /**
+     * Mutes / unmutes the player.
+     * @param mute true to mute the shared video, false - otherwise.
+     */
+    mutePlayer(mute) {
+        if (!this.player.isMuted() && mute) {
+            this.player.mute();
+            this.smartUnmute();
```

Comment:
I was trying to not use the player mute, cause it brings another complexity. As the player can be muted and put with volume to 0. So everywhere we were using just volume 0. Just on the checks where we need to send info for player state we do: this.player.isMuted() ? 0 : this.player.getVolume(). If its muted we just send volume 0, which produces muted player.